# Exploring eBay Car Sales Data


## Introduction
In this project, we'll work with a dataset of used cars from eBay Kleinanzeigen, a classifieds section of the German eBay website.

The dataset was originally scraped and uploaded to Kaggle by Dataquest.
The original dataset isn't available on Kaggle anymore, but you can find it [here](https://data.world/data-society/used-cars-data).


The **aim** of this project is to clean the data and analyze the included used car listings.


### Data dictionary ###

The data dictionary provided with data is as follows:

* `dateCrawled` - When this ad was first crawled. All field-values are taken from this date.
* `name` - Name of the car.
* `seller` - Whether the seller is private or a dealer.
* `offerType` - The type of listing
* `price` - The price on the ad to sell the car.
* `abtest` - Whether the listing is included in an A/B test.
* `vehicleType` - The vehicle Type.
* `yearOfRegistration` - The year in which the car was first registered.
* `gearbox` - The transmission type.
* `powerPS` - The power of the car in PS.
* `model` - The car model name.
* `kilometer` - How many kilometers the car has driven.
* `monthOfRegistration` - The month in which the car was first registered.
* `fuelType` - What type of fuel the car uses.
* `brand` - The brand of the car.
* `notRepairedDamage` - If the car has a damage which is not yet repaired.
* `dateCreated` - The date on which the eBay listing was created.
* `nrOfPictures` - The number of pictures in the ad.
* `postalCode` - The postal code for the location of the vehicle.
* `lastSeenOnline` - When the crawler saw this ad last online.





## Downloading the Dataset

In [1]:
import pandas as pd
import numpy as np

autos = pd.read_csv('autos.csv', encoding="Latin-1")
autos.head()
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

In [2]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  yearOfRegistration  \
0      privat   Angebot   $5,000  control         bus                2004   
1      privat   Angebot   $8,500  control   limousine                1997   
2      privat   Angebot   $8,990     test   limousine                2009   
3      privat   Angebot   $4,350  control  kleinwagen                2007   
4      privat   Angebot   $1,350     test       kombi                2003   
...       ...       ...      ...      ...         ...                 ...   
49995  privat   Angebot  $24,900  control   limousine                2011   
49996  privat   Angebot   $1,980  control      cabrio                1996   
49997  privat   Angebot  $13,200     test      cabrio                2014   
49998  privat   Angebot  $22,900  control       kombi                2013   
49999  privat   Angebot   $1,250  control   limousine                1996   

         gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0        manuell      158  andere  150,000km                    3      lpg   
1      automatik      286     7er  150,000km                    6   benzin   
2        manuell      102    golf   70,000km                    7   benzin   
3      automatik       71  fortwo   70,000km                    6   benzin   
4        manuell        0   focus  150,000km                    7   benzin   
...          ...      ...     ...        ...                  ...      ...   
49995  automatik      239      q5  100,000km                    1   diesel   
49996    manuell       75   astra  150,000km                    5   benzin   
49997  automatik       69     500    5,000km                   11   benzin   
49998    manuell      150      a3   40,000km                   11   diesel   
49999    manuell      101  vectra  150,000km                    1   benzin   

            brand notRepairedDamage          dateCreated  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4999

In [3]:
print(autos.info)

<bound method DataFrame.info of                dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V

Observations:
- Certain columns with date values are object columns like: lastSeen, dateCreated,dateCrawled
- The price column is also an object column
- 5 columns contain NULL data: notRepairedDamage, fuelType, model, gearbox, vehicleType


In [4]:
print(autos.head())

           dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere 

Further observations:
- The dataset contains 20 columns, most of which are strings.
- Some columns have null values, but none have more than ~20% null values.
- The column names use camelcase instead of Python's preferred snakecase, which means we can't just replace spaces with underscores.

## Cleaning Column Names

In [5]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

We will convert the column names from camelcase to Python style snakecase and rename some column names to make them more descriptive.

In [6]:
# renaming the column names
autos = autos.rename({'dateCrawled': 'date_crawled', 
                        'offerType': 'offer_type',
                        'vehicleType': 'vehicle_type',
                        'yearOfRegistration': 'registration_year',
                        'monthOfRegistration': 'registration_month', 
                        'powerPS': 'power_ps',
                        'odometer': 'used_kilometers',
                        'fuelType': 'fuel_type',
                        'notRepairedDamage': 'unrepaired_damage',
                        'dateCreated': 'ad_created',
                        'nrOfPictures': 'total_pictures', 
                        'postalCode': 'postal_code', 
                        'lastSeen': 'last_seen'}, axis=1)

print(autos.columns)

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'used_kilometers', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'total_pictures', 'postal_code',
       'last_seen'],
      dtype='object')


In [7]:
print(autos.head())

          date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model used_kilometers  registration_month fuel_type  \
0    manuell      

Now we have column names that are in line with the Python naming convention, which are also clear and descriptive.

## Initial Exploration and Cleaning

Exploring the data to find if the data needs to be cleaned. We will look for:

- Any columns that have mostly one value that are candidates to be dropped
- Any columns that need more investigation.
- Any examples of numeric data stored as text that needs to be cleaned.


In [8]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-29 23:42:13  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

       used_kilometers  registration_month fuel_type       brand  \
count            50000        50000.000000     45518       50000   
unique              13                 NaN         7          40   
top          150,000km                 NaN    benzin  volkswagen   
freq             32424                 NaN     30107       10687   
mean               NaN            5.723360       NaN         NaN   
std                NaN            3.711984       NaN         NaN   
min                NaN            0.000000       NaN         NaN   
25%                NaN            3.000000       NaN         NaN   
50%                NaN            6.000000       NaN         NaN   
75%                NaN            9.000000       NaN         NaN   
max                NaN           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  total_pictures   postal_code  \
count              40171                50000         50000.0  50000.000000   
unique                 2                   76             NaN           NaN   
top                 nein  2016-04-03 00:00:00             NaN           NaN   
freq               35232                 1946             NaN           NaN   
mean                 NaN                  NaN             0.0  50813.627300   
std                  NaN                  NaN             0.0  25779.747957   
min                  NaN                  NaN             0.0   1067.000000   
25%                  NaN                  NaN             0.0  30451.000000   
50%                  NaN                  NaN             0.0  49577.000000   
75%                  NaN                  NaN             0.0  71540.000000   
max                  NaN                  NaN             0.0  99998.000000   

                  last_seen  
count                 50000  
unique                39481  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN


### Observations:
* Columns that have mostly one value are `seller`, `offer_type` and `total_pictures`. The first two columns are text columns that contain only 2 unique values.
* All of the values in the column `total_pictures` are zero's.
* Columns that need more investigation because these columns contain outliers: `power_ps`, `registration_year`, `registration_month`.
* Columns that contain numeric data stored as text that needs cleaning and should be converted to an int data type: `used_kilometers`, `price`


Now we will drop the non-informative columns `seller`, `offer_type` and `total_pictures`.


In [9]:
# Dropping columns
autos.drop(['seller', 'offer_type', 'total_pictures'], axis=1, inplace=True)


Now we will clean the `price` and `used_kilometers` columns and convert them to an int datatype.

In [10]:
autos['price'] = autos['price'].str.replace('$', '').str.replace(',','').astype(int)
print(autos['price'].head)


<bound method NDFrame.head of 0         5000
1         8500
2         8990
3         4350
4         1350
         ...  
49995    24900
49996     1980
49997    13200
49998    22900
49999     1250
Name: price, Length: 50000, dtype: int64>


In [11]:
print((autos['used_kilometers'].unique), "\n")
autos['used_kilometers'] = autos['used_kilometers'].str.replace('km', '').str.replace(',','').astype(int)
print(autos['used_kilometers'].head)

<bound method Series.unique of 0        150,000km
1        150,000km
2         70,000km
3         70,000km
4        150,000km
           ...    
49995    100,000km
49996    150,000km
49997      5,000km
49998     40,000km
49999    150,000km
Name: used_kilometers, Length: 50000, dtype: object> 

<bound method NDFrame.head of 0        150000
1        150000
2         70000
3         70000
4        150000
          ...  
49995    100000
49996    150000
49997      5000
49998     40000
49999    150000
Name: used_kilometers, Length: 50000, dtype: int64>


## Exploring the Odometer and Price Columns

We will continue to analyze the `price` and `used_kilometers` columns. Looking for data that doesn't look right.

In [12]:
autos['price'].unique().shape

(2357,)

In [13]:
autos['used_kilometers'].unique().shape

(13,)

The `price` column has 2357 unique values and the `used_kilometers` column has 13 unique values.

Now we have a look at the statics of the `price` column.

In [14]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [15]:
# converting numbers to strings to make the output more readable
autos['price'].describe().apply(lambda x: format(x, 'f'))

count       50000.000000
mean         9840.043760
std        481104.380500
min             0.000000
25%          1100.000000
50%          2950.000000
75%          7200.000000
max      99999999.000000
Name: price, dtype: object

It seems that there is an outlier in prices: the maximum value 99999999 dollars is very high compared to the value of the third quartile (75%) 7200 dollars.

In [16]:
# Finding the 5 biggest prices in descending order
five_biggest_prices = autos['price'].value_counts().sort_index(ascending=False).head()
print(five_biggest_prices)


99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
Name: price, dtype: int64


The maximum value seems to be incorrect as the maximum price is an extreme outlier. Also it seems to be a very unreasonable price for a used car. Thusthis value should be removed.

In [17]:
# Removing the outlier from the price column
autos = autos.loc[autos['price'].between(0, 99999998), :]

In [18]:
autos['used_kilometers'].describe()

count     49999.000000
mean     125732.214644
std       40042.465064
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: used_kilometers, dtype: float64

In [19]:
used_km_values = autos['used_kilometers'].value_counts().sort_index(ascending=False)
used_km_values

150000    32423
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
40000       819
30000       789
20000       784
10000       264
5000        967
Name: used_kilometers, dtype: int64

This column doesn't have any outliers. The values range between the 5000 and 150.000 kilometers. And the dataset has a mean of 125.732 kilometers. The statistics in the previous cell show that the maximum value of 150.000 km has been driven by 65% of the cars.

In [20]:
autos['date_crawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=True)

2016-03-05    0.025381
2016-03-06    0.013940
2016-03-07    0.035961
2016-03-08    0.033301
2016-03-09    0.033221
2016-03-10    0.032121
2016-03-11    0.032481
2016-03-12    0.036781
2016-03-13    0.015560
2016-03-14    0.036621
2016-03-15    0.033981
2016-03-16    0.029501
2016-03-17    0.031521
2016-03-18    0.013060
2016-03-19    0.034901
2016-03-20    0.037821
2016-03-21    0.037521
2016-03-22    0.032921
2016-03-23    0.032381
2016-03-24    0.029101
2016-03-25    0.031741
2016-03-26    0.032481
2016-03-27    0.031041
2016-03-28    0.034841
2016-03-29    0.034181
2016-03-30    0.033621
2016-03-31    0.031921
2016-04-01    0.033801
2016-04-02    0.035401
2016-04-03    0.038681
2016-04-04    0.036521
2016-04-05    0.013100
2016-04-06    0.003180
2016-04-07    0.001420
Name: date_crawled, dtype: float64

#### Date_crawled column:
The data has been crawled from the website between 05-03-2016 and 07-04-2016.

In [21]:
# Extracting days, counting values in % including missing values, and sorting the values

ads_created = autos['ad_created'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=True)
print(ads_created)

2015-06-11    0.000020
2015-08-10    0.000020
2015-09-09    0.000020
2015-11-10    0.000020
2015-12-05    0.000020
                ...   
2016-04-03    0.038921
2016-04-04    0.036881
2016-04-05    0.011840
2016-04-06    0.003260
2016-04-07    0.001280
Name: ad_created, Length: 76, dtype: float64


#### Ad_created column:
The data shows that the ads were created between 11-06-2015 and 07-04-2016.

In [22]:
# Extracting days, counting values in % including missing values, and sorting the values

last_seen = autos['last_seen'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=True)
print(last_seen)

2016-03-05    0.001080
2016-03-06    0.004420
2016-03-07    0.005360
2016-03-08    0.007600
2016-03-09    0.009860
2016-03-10    0.010760
2016-03-11    0.012520
2016-03-12    0.023820
2016-03-13    0.008980
2016-03-14    0.012800
2016-03-15    0.015880
2016-03-16    0.016440
2016-03-17    0.027921
2016-03-18    0.007420
2016-03-19    0.015740
2016-03-20    0.020700
2016-03-21    0.020740
2016-03-22    0.021580
2016-03-23    0.018580
2016-03-24    0.019560
2016-03-25    0.019200
2016-03-26    0.016960
2016-03-27    0.016020
2016-03-28    0.020860
2016-03-29    0.022340
2016-03-30    0.024840
2016-03-31    0.023840
2016-04-01    0.023100
2016-04-02    0.024900
2016-04-03    0.025361
2016-04-04    0.024620
2016-04-05    0.124282
2016-04-06    0.220984
2016-04-07    0.130923
Name: last_seen, dtype: float64


#### Last_seen column:

The dates in this column range from 2016-03-05 until 2016-04-07. The values seem to be uniform, except for the last three days a lot more advertisements have been seen for the last time online, which means a lot more sales have been made in these days.

## Cleaning Registration Year Data

The registration_year column contains some odd values:
- The minimum value is 1000, before cars were invented
- The maximum value is 9999, many years into the future

Because a car can't be first registered after the listing was seen, any vehicle with a registration year above 2016 is definitely inaccurate. Determining the earliest valid year is more difficult. Realistically, it could be somewhere in the first few decades of the 1900s.

In [23]:
autos['registration_year'].describe()

count    49999.000000
mean      2005.073401
std        105.713866
min       1000.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

In [24]:
registration_year = autos['registration_year'].value_counts().sort_index()
print(registration_year)

1000    1
1001    1
1111    1
1500    1
1800    2
       ..
6200    1
8888    1
9000    2
9996    1
9999    4
Name: registration_year, Length: 97, dtype: int64


In [25]:
autos = autos[autos['registration_year'].between(1910,2016)]

Reasoning: Because a car can't be first registered after the listing was seen, any vehicle with a registration year above 2016 is definitely inaccurate. So the cars with a registration year above 2016 have been removed. And before the year 1910, 6 cars have been registered so we removed these too.

In [26]:
autos['registration_year'].value_counts(normalize=True).sort_index(ascending=False)

2016    0.027401
2015    0.008308
2014    0.013867
2013    0.016782
2012    0.027547
          ...   
1934    0.000042
1931    0.000021
1929    0.000021
1927    0.000021
1910    0.000187
Name: registration_year, Length: 78, dtype: float64

#### Observations:
Most cars in this dataset list from the period 1994 to 2016.
The frequencies of car registration years before the year 1994 are relatively low compared to more recent years.

## Exploring Price by Brand

The next step is to explore price varations accross the different car brands. First we will have a look at the unique brands in this dataset.

In [27]:
autos['brand'].unique()

array(['peugeot', 'bmw', 'volkswagen', 'smart', 'ford', 'chrysler',
       'seat', 'renault', 'mercedes_benz', 'audi', 'sonstige_autos',
       'opel', 'mazda', 'porsche', 'mini', 'toyota', 'dacia', 'nissan',
       'jeep', 'saab', 'volvo', 'mitsubishi', 'jaguar', 'fiat', 'skoda',
       'subaru', 'kia', 'citroen', 'chevrolet', 'hyundai', 'honda',
       'daewoo', 'suzuki', 'trabant', 'land_rover', 'alfa_romeo', 'lada',
       'rover', 'daihatsu', 'lancia'], dtype=object)

In [28]:

# counts values in the 'brand' column
top_9_brands = autos['brand'].value_counts().index[:9]
print(top_9_brands)


Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat'],
      dtype='object')


I have chosen to aggregate the data on the top 9 cars, that have a value over 1000. 

In [29]:

# Creating an empty dictionary for the mean price per top brand
mean_price_top_9_cars = {}

# Looping over the top brands and assiging the average price to the dictionary
for brand in top_9_brands:
    mean_price_top_9_cars[brand] = autos[autos["brand"] == brand]['price'].mean()

mean_price_top_9_cars

{'volkswagen': 6516.457597173145,
 'bmw': 8334.645155185466,
 'opel': 5252.61655437921,
 'mercedes_benz': 8485.239571958942,
 'audi': 9093.65003615329,
 'ford': 7263.015811455847,
 'renault': 2395.4164467897976,
 'peugeot': 3039.4682651622,
 'fiat': 2711.8011272141707}

In [30]:
# Creating a series object of the dictionary
mean_price_top_9_cars_series = pd.Series(mean_price_top_9_cars)
mean_price_top_9_cars_series = mean_price_top_9_cars_series.sort_values(ascending=False)
print(mean_price_top_9_cars_series)


# creating a Dataframe object

Overview_brands = pd.DataFrame(mean_price_top_9_cars_series, columns=['mean_price'])

audi             9093.650036
mercedes_benz    8485.239572
bmw              8334.645155
ford             7263.015811
volkswagen       6516.457597
opel             5252.616554
peugeot          3039.468265
fiat             2711.801127
renault          2395.416447
dtype: float64


The top 3 cars with the highest mean price are: audi with a price of 9093.65, mercedes_benz with a price of 8485.23 and then bmw with a price of 8334.64. The cheapest brands are: renault, fiat and peugeot.
And in between prices: opel, volkswagen and ford.

In [31]:

# Creating an empty dictionary
mean_mileage_per_top_brands = {}

# Looping over the top brands and assigning the mean average mileage to the dictionary
for brand in top_9_brands:
    mean_mileage_per_top_brands[brand] = autos[autos["brand"] == brand]['used_kilometers'].mean()

mean_mileage_per_top_brands

{'volkswagen': 128730.36906164115,
 'bmw': 132434.70855412565,
 'opel': 129227.14148219442,
 'mercedes_benz': 130856.0821139987,
 'audi': 129287.78018799711,
 'ford': 124046.83770883054,
 'renault': 128183.81706244503,
 'peugeot': 127136.81241184767,
 'fiat': 116553.94524959743}

In [32]:
# Creating a series object from the dictionary
mean_mileage_series = pd.Series(mean_mileage_per_top_brands)
mean_mileage_series = mean_mileage_series.sort_values(ascending=False)
print(mean_mileage_series)

Overview_brands["mean_mileage"] = mean_mileage_series

bmw              132434.708554
mercedes_benz    130856.082114
audi             129287.780188
opel             129227.141482
volkswagen       128730.369062
renault          128183.817062
peugeot          127136.812412
ford             124046.837709
fiat             116553.945250
dtype: float64


In [33]:
print(Overview_brands)

                mean_price   mean_mileage
audi           9093.650036  129287.780188
mercedes_benz  8485.239572  130856.082114
bmw            8334.645155  132434.708554
ford           7263.015811  124046.837709
volkswagen     6516.457597  128730.369062
opel           5252.616554  129227.141482
peugeot        3039.468265  127136.812412
fiat           2711.801127  116553.945250
renault        2395.416447  128183.817062


In the combined pandas dataframe we can see that the mean_price column and mean_mileage columns are not directly related to each other. The mileage doesn't seem to have a big impact on the price, and is dependent on other factors as well as the condition, model and year of the car.